src/ripple/rpc/handlers/ServerInfo.cpp

```diff
@@ -22,13 +22,313 @@
+            v[jss::isSerialized] = false;
+            v[jss::isSigningField] = false;
+            v[jss::type] = "Amount";
+            a[1U] = v;
+            ret[jss::FIELDS][i++] = a;
+        }
+
+        for (auto const& [code, f] : ripple::SField::knownCodeToField)
+        {
+            if (f->fieldName == "")
+                continue;
+
+            Json::Value innerObj = Json::objectValue;
+
+            uint32_t type = f->fieldType;
+
+            innerObj[jss::nth] = f->fieldValue;
+
+            innerObj[jss::isVLEncoded] =
+                (type == 7U /* Blob       */ || type == 8U /* AccountID  */ ||
+                 type == 19U /* Vector256  */);
+
+            innerObj[jss::isSerialized] =
+                (type < 10000 && f->fieldName != "hash" &&
+                 f->fieldName !=
+                     "index"); /* hash, index, TRANSACTION, LEDGER_ENTRY,
+                                  VALIDATION, METADATA */
+
+            innerObj[jss::isSigningField] = f->shouldInclude(false);
+
+            innerObj[jss::type] = type_map[type];
+
+            Json::Value innerArray = Json::arrayValue;
+            innerArray[0U] = f->fieldName;
+            innerArray[1U] = innerObj;
+
+            ret[jss::FIELDS][i++] = innerArray;
+        }
+
+        ret[jss::TRANSACTION_RESULTS] = Json::objectValue;
+
+        for (const auto& pair : detail::transResults())
+        {
+            ret[jss::TRANSACTION_RESULTS][str(pair.second.first)] =
+                std::stoi(str(pair.first));
+        }
+
+        ret[jss::TRANSACTION_TYPES] = Json::objectValue;
+        ret[jss::TRANSACTION_TYPES][jss::Invalid] = -1;
+        for (const auto& pair : TxFormats::getInstance().getNamesAndTypes())
+            ret[jss::TRANSACTION_TYPES][str(pair.first)] =
+                std::stoi(str(pair.second));
+
+        // generate hash
+        {
+            const std::string out = Json::FastWriter().write(ret);
+            defsHash =
+                ripple::sha512Half(ripple::Slice{out.data(), out.size()});
+            ret[jss::hash] = to_string(*defsHash);
+        }
+
+        return ret;
+    }
+
+    std::optional<uint256> defsHash;
+    Json::Value defs;
+
+public:
+    Definitions() : defs(generate()){};
+
+    bool
+    hashMatches(uint256 hash) const
+    {
+        return defsHash && *defsHash == hash;
+    }
+
+    Json::Value const&
+    operator()(void) const
+    {
+        return defs;
+    }
+};
+
+Json::Value
+doServerDefinitions(RPC::JsonContext& context)
+{
+    auto& params = context.params;
+
+    uint256 hash;
+    if (params.isMember(jss::hash))
+    {
+        if (!params[jss::hash].isString() ||
+            !hash.parseHex(params[jss::hash].asString()))
+            return RPC::invalid_field_error(jss::hash);
+    }
+
+    static const Definitions defs{};
+    if (defs.hashMatches(hash))
+    {
+        Json::Value jv = Json::objectValue;
+        jv[jss::hash] = to_string(hash);
+        return jv;
```

Comment:
Most of the code changes in this file were written by @RichardAH in #4402 so I haven't looked over it in detail.

Reply:
The output does seem to have a `"status": "success"` element when I run the command locally.

Reply:
When you query the API, if you have a pre-existing set of definitions then you supply the hash of those definitions. If the server has the same definitions it just sends back the hash again to confirm, rather than the whole definitions. If the server has different definitions then it sends those plus the new hash.

Reply:
It looks like the `"status": "success"` element is added automatically to RPCs and doesn't need to be done in the handlers themselves: https://github.com/XRPLF/rippled/blob/develop/src/ripple/rpc/impl/ServerHandler.cpp#L531